test/e2e/set/scylladbdatacenter/scylladbdatacenter_globalmanager.go

```diff
@@ -0,0 +1,191 @@
+	"github.com/scylladb/scylla-operator/test/e2e/utils"
+	utilsv1alpha1 "github.com/scylladb/scylla-operator/test/e2e/utils/v1alpha1"
+	scylladbdatacenterverification "github.com/scylladb/scylla-operator/test/e2e/utils/verification/scylladbdatacenter"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = g.Describe("ScyllaDBDatacenter integration with global ScyllaDB Manager", func() {
+	f := framework.NewFramework("scylladbdatacenter")
+
+	hasDeletionFinalizer := func(smcr *scyllav1alpha1.ScyllaDBManagerClusterRegistration) (bool, error) {
+		return slices.ContainsItem(smcr.Finalizers, naming.ScyllaDBManagerClusterRegistrationFinalizer), nil
+	}
+
+	g.It("should register labeled ScyllaDBDatacenter and deregister it when it's unlabeled", func(ctx g.SpecContext) {
+		ns, nsClient, ok := f.DefaultNamespaceIfAny()
+		o.Expect(ok).To(o.BeTrue())
+
+		sdc := f.GetDefaultScyllaDBDatacenter()
+		metav1.SetMetaDataLabel(&sdc.ObjectMeta, naming.GlobalScyllaDBManagerRegistrationLabel, naming.LabelValueTrue)
+
+		framework.By(`Creating a ScyllaDBDatacenter with the global ScyllaDB Manager registration label`)
+		sdc, err := nsClient.ScyllaClient().ScyllaV1alpha1().ScyllaDBDatacenters(ns.Name).Create(ctx, sdc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for ScyllaDBDatacenter to roll out (RV=%s)", sdc.ResourceVersion)
+		rolloutCtx, rolloutCtxCancel := utilsv1alpha1.ContextForRollout(ctx, sdc)
+		defer rolloutCtxCancel()
+		sdc, err = controllerhelpers.WaitForScyllaDBDatacenterState(rolloutCtx, nsClient.ScyllaClient().ScyllaV1alpha1().ScyllaDBDatacenters(ns.Name), sdc.Name, controllerhelpers.WaitForStateOptions{}, utilsv1alpha1.IsScyllaDBDatacenterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		scylladbdatacenterverification.Verify(ctx, nsClient.KubeClient(), nsClient.ScyllaClient(), sdc)
+		scylladbdatacenterverification.WaitForFullQuorum(ctx, nsClient.KubeClient().CoreV1(), sdc)
+
+		hosts, err := utilsv1alpha1.GetBroadcastRPCAddresses(ctx, f.KubeClient().CoreV1(), sdc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(hosts).To(o.HaveLen(1))
+		di := scylladbdatacenterverification.InsertAndVerifyCQLData(ctx, hosts)
+		defer di.Close()
+
+		smcrName, err := naming.ScyllaDBManagerClusterRegistrationNameForScyllaDBDatacenter(sdc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for ScyllaDBDatacenter to register with global ScyllaDB Manager instance")
+		registrationCtx, registrationCtxCancel := context.WithTimeout(ctx, utils.SyncTimeout)
+		defer registrationCtxCancel()
+		smcr, err := controllerhelpers.WaitForScyllaDBManagerClusterRegistrationState(registrationCtx, nsClient.ScyllaClient().ScyllaV1alpha1().ScyllaDBManagerClusterRegistrations(ns.Name), smcrName, controllerhelpers.WaitForStateOptions{}, utilsv1alpha1.IsScyllaDBManagerClusterRegistrationRolledOut, hasDeletionFinalizer)
```

Comment:
lets break down this line in a way that conditions we are awaiting for are in separate lines

Reply:
done